uat/custom-components/client-python-paho/src/grpc_client_server/grpc_control_server.py

```diff
@@ -103,7 +119,7 @@ async def ShutdownAgent(
         self.unblock_wait()
         return Empty()
 
-    async def CreateMqttConnection(
+    async def CreateMqttConnection(  # pylint: disable=too-many-locals,too-many-branches,too-many-statements

```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This function contains 54 lines of code, not including blank lines or lines with only comments, Python punctuation characters, identifiers, or literals. By comparison, 98% of the functions in the CodeGuru reference dataset contain fewer lines of code. Large functions might be difficult to read and have logic that is hard to understand and test.

We recommend that you simplify this function or break it into multiple functions. For example, consider extracting the code block on lines 185-241 into a separate function.

Reply:
fixed in c1b9703